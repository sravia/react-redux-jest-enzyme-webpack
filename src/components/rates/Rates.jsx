import React from 'react';
import PropTypes from 'prop-types';

import './style.scss';

const Rates = ({ rates }) => (
  <div className="rates">
    <table>
      <tbody>
        {Object.keys(rates).map(key => (
          <tr key={key}>
            <td>{key}</td>
            <td>{rates[key]}</td>
          </tr>
        ))}
      </tbody>
    </table>
  </div>
);

Rates.propTypes = {
  rates: PropTypes.object.isRequired
};

export default Rates;
