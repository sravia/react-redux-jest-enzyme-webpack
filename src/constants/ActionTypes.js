export const GET_RATES_REQUEST = 'GET_RATES_REQUEST';
export const GET_RATES_SUCCESS = 'GET_RATES_SUCCESS';
export const GET_RATES_FAILURE = 'GET_RATES_FAILURE';
