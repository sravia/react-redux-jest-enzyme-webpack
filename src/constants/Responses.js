export const API_ERROR = 'Could not fetch latest rates!';
export const TEST_API_RESPONSE = {
  date: '2018-10-23',
  rates: {
    BGN: 1.9558,
    CAD: 1.5047,
    BRL: 4.2412,
    HUF: 322.81,
    DKK: 7.4603,
    JPY: 128.76,
    ILS: 4.2086,
    TRY: 6.6227,
    RON: 4.667,
    GBP: 0.88103,
    PHP: 61.716,
    HRK: 7.4275,
    NOK: 9.5053,
    USD: 1.1478,
    MXN: 22.2469,
    AUD: 1.6243,
    IDR: 17435.02,
    KRW: 1305.81,
    HKD: 8.999,
    ZAR: 16.4759,
    ISK: 136.3,
    CZK: 25.814,
    THB: 37.705,
    MYR: 4.7778,
    NZD: 1.7514,
    PLN: 4.296,
    SEK: 10.3606,
    RUB: 75.0301,
    CNY: 7.9613,
    SGD: 1.5822,
    CHF: 1.1424,
    INR: 84.4435
  },
  base: 'EUR'
};
