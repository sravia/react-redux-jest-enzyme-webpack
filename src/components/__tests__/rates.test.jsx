import React from 'react';
import renderer from 'react-test-renderer';
import Rates from '../rates/Rates';
import { TEST_API_RESPONSE } from '../../constants/responses';

describe('<Rates />', () => {
  it('renders <Rates /> component', () => {
    const tree = renderer
      .create(<Rates rates={TEST_API_RESPONSE.rates} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
