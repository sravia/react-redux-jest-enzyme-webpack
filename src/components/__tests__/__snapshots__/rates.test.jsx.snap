// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Rates /> renders <Rates /> component 1`] = `
<div
  className="rates"
>
  <table>
    <tbody>
      <tr>
        <td>
          BGN
        </td>
        <td>
          1.9558
        </td>
      </tr>
      <tr>
        <td>
          CAD
        </td>
        <td>
          1.5047
        </td>
      </tr>
      <tr>
        <td>
          BRL
        </td>
        <td>
          4.2412
        </td>
      </tr>
      <tr>
        <td>
          HUF
        </td>
        <td>
          322.81
        </td>
      </tr>
      <tr>
        <td>
          DKK
        </td>
        <td>
          7.4603
        </td>
      </tr>
      <tr>
        <td>
          JPY
        </td>
        <td>
          128.76
        </td>
      </tr>
      <tr>
        <td>
          ILS
        </td>
        <td>
          4.2086
        </td>
      </tr>
      <tr>
        <td>
          TRY
        </td>
        <td>
          6.6227
        </td>
      </tr>
      <tr>
        <td>
          RON
        </td>
        <td>
          4.667
        </td>
      </tr>
      <tr>
        <td>
          GBP
        </td>
        <td>
          0.88103
        </td>
      </tr>
      <tr>
        <td>
          PHP
        </td>
        <td>
          61.716
        </td>
      </tr>
      <tr>
        <td>
          HRK
        </td>
        <td>
          7.4275
        </td>
      </tr>
      <tr>
        <td>
          NOK
        </td>
        <td>
          9.5053
        </td>
      </tr>
      <tr>
        <td>
          USD
        </td>
        <td>
          1.1478
        </td>
      </tr>
      <tr>
        <td>
          MXN
        </td>
        <td>
          22.2469
        </td>
      </tr>
      <tr>
        <td>
          AUD
        </td>
        <td>
          1.6243
        </td>
      </tr>
      <tr>
        <td>
          IDR
        </td>
        <td>
          17435.02
        </td>
      </tr>
      <tr>
        <td>
          KRW
        </td>
        <td>
          1305.81
        </td>
      </tr>
      <tr>
        <td>
          HKD
        </td>
        <td>
          8.999
        </td>
      </tr>
      <tr>
        <td>
          ZAR
        </td>
        <td>
          16.4759
        </td>
      </tr>
      <tr>
        <td>
          ISK
        </td>
        <td>
          136.3
        </td>
      </tr>
      <tr>
        <td>
          CZK
        </td>
        <td>
          25.814
        </td>
      </tr>
      <tr>
        <td>
          THB
        </td>
        <td>
          37.705
        </td>
      </tr>
      <tr>
        <td>
          MYR
        </td>
        <td>
          4.7778
        </td>
      </tr>
      <tr>
        <td>
          NZD
        </td>
        <td>
          1.7514
        </td>
      </tr>
      <tr>
        <td>
          PLN
        </td>
        <td>
          4.296
        </td>
      </tr>
      <tr>
        <td>
          SEK
        </td>
        <td>
          10.3606
        </td>
      </tr>
      <tr>
        <td>
          RUB
        </td>
        <td>
          75.0301
        </td>
      </tr>
      <tr>
        <td>
          CNY
        </td>
        <td>
          7.9613
        </td>
      </tr>
      <tr>
        <td>
          SGD
        </td>
        <td>
          1.5822
        </td>
      </tr>
      <tr>
        <td>
          CHF
        </td>
        <td>
          1.1424
        </td>
      </tr>
      <tr>
        <td>
          INR
        </td>
        <td>
          84.4435
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
